src/main/java/org/killbill/billing/plugin/adyen/client/payment/service/AdyenCheckoutApiClient.java

```diff
@@ -27,13 +35,76 @@ public AdyenCheckoutApiClient(final AdyenConfigProperties adyenConfigProperties)
         checkoutApi = new Checkout(client);
     }
 
-    public PaymentsResponse createPayment(PaymentsRequest request) throws ApiException, IOException {
-        PaymentsResponse response = checkoutApi.payments(request);
-        return response;
+    public AdyenCallResult<PaymentsResponse> createPayment(PaymentsRequest request) {
+        final AdyenCallResult<PaymentsResponse> result;
+        result = callApi(request, new ApiRequest<PaymentsRequest, PaymentsResponse>() {
+            @Override
+            public PaymentsResponse call() throws ApiException, IOException {
+                return checkoutApi.payments(request);
+            }
+        });
+
+        return result;
+    }
+
+    public AdyenCallResult<PaymentsResponse> paymentDetails(PaymentsDetailsRequest request) {
+        final AdyenCallResult<PaymentsResponse> result;
+        result = callApi(request, new ApiRequest<PaymentsDetailsRequest, PaymentsResponse>() {
+            @Override
+            public PaymentsResponse call() throws ApiException, IOException {
+                return checkoutApi.paymentsDetails(request);
+            }
+        });
+
+        return result;
+    }
+
+    private <REQ, RES> AdyenCallResult<RES> callApi(REQ request, ApiRequest<REQ, RES> apiRequest) {
+        final String logRequest = request.toString();
+        logger.info("Checkout API request:" + logRequest);
+
+        final long startTime = System.currentTimeMillis();
+        try {
+            final RES result = apiRequest.call();
+            final long duration = System.currentTimeMillis() - startTime;
+            final String logResult = result.toString();
+            logger.info("Checkout API duration:"+ duration +" response: " + logResult);
+            return new SuccessfulAdyenCall<RES>(result, duration);
+        } catch (ApiException ex) {
+            final long duration = System.currentTimeMillis() - startTime;
+            logger.warn("Exception during Adyen request", ex);
+            return handleException(ex, duration);
+        } catch (IOException ex) {
+            final long duration = System.currentTimeMillis() - startTime;
+            logger.warn("Exception during Adyen request", ex);
+            return handleException(ex, duration);
+        }
+    }
+
+    private <T> UnSuccessfulAdyenCall<T> handleException(final Exception ex, final long duration) {
+        final Throwable rootCause = Throwables.getRootCause(ex);
```

Comment:
why are we passing duration if we are not using it

Reply:
Added the duration logs. It will be useful for knowing the latency issues from Adyen API.

Reply:
+1 @harendrakr